runtime/src/main/java/org/corfudb/runtime/collections/Streamer.java

```diff
@@ -0,0 +1,46 @@
+package org.corfudb.runtime.collections;
+
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * This is the callback interface that any client subscribe()ing to CorfuStore updates must implement.
+ *
+ * Created by hisundar on 2019-10-18
+ */
+public interface Streamer {
+    /**
+     * A corfu update may have multiple updates belonging to different streams.
+     * This callback will return those updates as a list grouped by their Stream UUIDs.
+     *
+     * @param results is a map of stream UUID -> list of entries of this stream.
+     */
+    void onNext(Map<UUID, List<CorfuStreamEntry>> results);
```

Comment:
We should hide the map data structure behind a result object of our own.
Let me know your thoughts.

Reply:
good idea!